### PR TITLE
(QA-2501) Make killing procs on exception optional, add timeout

### PR DESCRIPTION
**Pull request by samwoods1**

### CODE REVIEW

**Comment by samwoods1** [comment]
I made another commit on this PR.

File: lib/in_parallel.rb [hunk 1]
```diff
@@ -226,14 +240,53 @@ def self._execute_in_parallel(method_sym, obj = self, &block)
...
     end
 
+    def self.fork_supported?
+      @@supported ||= Process.respond_to?(:fork)
+      @@logger.warn 'Warning: Fork is not supported on this OS, executing block normally' unless @@supported
+      @@supported
+    end
+
+    def self.kill_child_processes
+      @@process_infos.each { |process_info|
```
Comment (zreichert) [inline lib/in_parallel.rb]: Multi line block should be `do ... end`

File: lib/in_parallel.rb [hunk 2]
```diff
@@ -226,14 +240,53 @@ def self._execute_in_parallel(method_sym, obj = self, &block)
...
+      }
+    end
+    private_class_method :kill_child_processes
+
+    # Private method to lookup results from the results_map and replace the
+    # temp values with actual return values
+    def self.result_lookup(proxy_obj, target_obj, results_map)
+      target_obj = eval('self', target_obj)
+      proxy_obj ||= target_obj
+      vars = (proxy_obj.instance_variables)
```
Comment (zreichert) [inline lib/in_parallel.rb]: Consistency for `=` method and parenthesis.

File: lib/in_parallel.rb [hunk 3]
```diff
@@ -226,14 +240,53 @@ def self._execute_in_parallel(method_sym, obj = self, &block)
...
+      results = []
+      results_map.each { |tmp_result|
+        results << tmp_result.values[0]
+        vars.each {|var|
```
Comment (zreichert) [inline lib/in_parallel.rb]: Multi line block should be `do ... end`

**Comment by zreichert** [comment]
@samwoods1 other than the style comments LTGM 👍

**Pull request merged**